### ISSUE TITLE
CloudFormation template for deployment

### ISSUE THREAD

**Issue by astrostl** [strofimovsky/aws-rolling-snaps#2]
I think this would be handy. Happy to tackle it myself.

**Comment by strofimovsky** [comment]
This would be great, thank you!

**Comment by astrostl** [comment]
I was thinking about this today. Will do soon!

**Comment by astrostl** [comment]
Started working this today with the new YAML syntax.

I'm close. There are some tradeoffs, though, like the need to S3-upload a zip file. Will have to be a wrapper script, and not JUST a CFN template.

**Comment by astrostl** [comment]
https://gist.github.com/astrostl/817d4d79161aa5ee4bee4014d3057241

**Comment by strofimovsky** [comment]
Sorry for delay, I was cleaning some pileups. Do you want to re-implement it with the introduction of lambda env and #13?

**Comment by astrostl** [comment]
No sweat. Was wondering if you were still around. Do you have a Twitter account?

Will get to CFN on the updates next week, most likely.

**Comment by strofimovsky** [comment]
I'm still around, just some work stuff piled up. I'm @troff on Twitter.

**Comment by alizaidis** [comment]
Chatted with @astrostl to implement the CFN template with Lambda environment variables. For review: https://gist.github.com/alizaidis/3fe467992dd4ee4e4914d7cfedf2df66

The .zip containing makesnap3.py needs to be in an S3 bucket and the CFN template needs to be kicked off. Interestingly, the stack fails to build due to this error:

```
The config profile (default) could not be found: ProfileNotFound
Traceback (most recent call last):
File "/var/task/makesnap3.py", line 277, in lambda_handler
return main(period)
File "/var/task/makesnap3.py", line 207, in main
config['aws_profile_name'] or 'default'))
File "/var/runtime/boto3/__init__.py", line 34, in setup_default_session
DEFAULT_SESSION = Session(**kwargs)
File "/var/runtime/boto3/session.py", line 80, in __init__
self._setup_loader()
File "/var/runtime/boto3/session.py", line 120, in _setup_loader
self._loader = self._session.get_component('data_loader')
File "/var/runtime/botocore/session.py", line 691, in get_component
return self._components.get_component(name)
File "/var/runtime/botocore/session.py", line 885, in get_component
self._components[name] = factory()
File "/var/runtime/botocore/session.py", line 179, in <lambda>
lambda: create_loader(self.get_config_variable('data_path')))
File "/var/runtime/botocore/session.py", line 259, in get_config_variable
elif self._found_in_config_file(methods, var_config):
File "/var/runtime/botocore/session.py", line 280, in _found_in_config_file
return var_config[0] in self.get_scoped_config()
File "/var/runtime/botocore/session.py", line 352, in get_scoped_config
raise ProfileNotFound(profile=profile_name)
ProfileNotFound: The config profile (default) could not be found
```

My specific use-case required exclusively using Lambda environment variables (thus no config.json) so I tried commenting out lines 205 ~ 207 in makesnap3.py:

```
if config.get('aws_profile_name'):
        boto3.setup_default_session(profile_name=(
            config['aws_profile_name'] or 'default'))
```

This seems to work without any issues.

**Comment by ibratoev** [comment]
I had a look into your CFN template. An important issue is that environment variable names should start with 'MAKESNAP_' ;)

**Comment by ibratoev** [comment]
I would also remove the `aws_profile_name` variable as it is not relevant for the Lambda scenario.

**Comment by ibratoev** [comment]
Additionally, parsing the string 'False' for a boolean variable doesn't work very well ;) You can use 1 and 0, example:
```
  SkipDelete:
    Type: Number
    Default: 0
```

`Edit`: Actually, that doesn't work as well because any string is Python is converted to `True`, except the empty string.

So to update the example above, all boolean parameters should be:
```
  SkipDelete:
    Type: String
    Default: ""
```

**Comment by alizaidis** [comment]
Thanks for reviewing the gist, and appreciate the feedback! I've updated the gist here; https://gist.github.com/alizaidis/3fe467992dd4ee4e4914d7cfedf2df66.

**Comment by strofimovsky** [comment]
Thank you guys for the template! Would you be able to create a pull request for it?

**Comment by alizaidis** [comment]
Happy to help! PR created.

**Comment by strofimovsky** [comment]
Merged in #14
